server/implementation/src/main/java/io/smallrye/graphql/execution/ExecutionService.java

```diff
@@ -77,11 +79,12 @@ public ExecutionResponse execute(JsonObject jsonInput) {
 
         try {
             String query = context.getQuery();
+            Optional<Map<String, Object>> variables = context.getVariables();
 
             if (query == null || query.isEmpty()) {
                 throw new RuntimeException("Query can not be null");
             }
-            if (config.logPayload()) {
+            if (config.logPayload().equals(Config.logPayloadOption.TRUE)) {
```

Comment:
So we have 3 scenarios. 
1) WITHOUTVARIABLE (and TRUE) - This needs to log the payload like it used to
2) WITHVARIABLE - This needs to log the payload and not also the variables (if any)
3) OFF (and FALSE) - Don't log anything.